src/background.js

```diff
@@ -173,36 +173,30 @@ const messageOps = {
 function syncBookmarks(items) {
   const title = browser.i18n.getMessage('bookmarkFolderTitle');
   browser.bookmarks.search({title: title}).then(folders => {
-    return Promise.all(folders.map(folder => {
-      return browser.bookmarks.removeTree(folder.id);
-    }));
-  }).then(() => {
+    if (folders.length) {
+      return folders[0];
+    }
     return browser.bookmarks.create({title: title});
   }).then(snoozeTabsFolder => {
     log('Sync Folder!', snoozeTabsFolder, Object.values(items));
-    const sortedEntries = [...Object.values(items)];
-    sortedEntries.sort((a, b) => {
-      if (a.time === NEXT_OPEN) {
-        if (b.time === NEXT_OPEN) {
-          return -(a.title.localeCompare(b.title));
-        } else {
-          return 1;
-        }
-      } else if (b.time === NEXT_OPEN) {
-        return -1;
-      } else {
-        return b.time - a.time;
-      }
-    });
+    return browser.bookmarks.getChildren(snoozeTabsFolder.id).then((bookmarks) => {
+      const tabs = [...Object.values(items)];
+      const toCreate = tabs.filter((tab) => !bookmarks.find((bookmark) => tab.url === bookmark.url));
+      const toRemove = bookmarks.filter((bookmark) => !tabs.find((tab) => bookmark.url === tab.url));
```

Comment:
"Kinda" indeed — this still won't work correctly with Sync.

Device A snoozes tabs AAAA, BBBB, and CCCC.
Device B syncs, so its Snoozed Tabs folder now contains AAAA, BBBB, and CCCC. But its snoozed tabs set (`items` here) is DDDD, EEEE. It'll now _efficiently and carefully_ remove AAAA, BBBB, and CCCC, insert DDDD and EEEE, and sync those up to the server, blowing away Device A's backup next time it syncs bookmarks. The two devices will fight over the folder each time they sync.